test/scripts/setup-osbuild-repo

```diff
@@ -3,27 +3,50 @@
 # Add a repository configuration to install the osbuild rpm at a specific
 # commit if specified.
 import os
+import sys
+import urllib.error
+import urllib.request
 
 import imgtestlib as testlib
 
 REPO_TEMPLATE = """
 [osbuild]
 name=osbuild {commit}
-baseurl=http://osbuild-composer-repos.s3-website.us-east-2.amazonaws.com/{repo_path}
+baseurl={baseurl}
 enabled=1
 gpgcheck=0
 priority=10
 """
 
 REPO_FILE = "/etc/yum.repos.d/osbuild.repo"
+URL_TEMPLATE = "http://osbuild-composer-repos.s3-website.us-east-2.amazonaws.com/{repo_path}"
+
+
+def baseurl(repo_path):
+    return URL_TEMPLATE.format(repo_path=repo_path)
+
+
+def check_baseurl(repo_baseurl):
+    repomd_url = repo_baseurl + "/repodata/repomd.xml"
+    try:
+        with urllib.request.urlopen(repomd_url, timeout=3) as resp:
```

Comment:
3 seems awfully short. If it succeeds it should be fast, if it is slow (this is hitting s3 after all) then it's going to fail anyway. I'd use (waving hands) 15 or 30 for good measure.

Reply:
I figured it'd be pretty fast since it's just doing a quick check for a URL request but you're right, S3 can be _quirky_.  Set to 30.